org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardPageRepositoryAndCredentials.java

```diff
@@ -316,11 +320,11 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
 
 			//if credentials are provided, ask user if the credentials should be stored in the secure storage
 			if (this.cloneData.user != null && this.cloneData.pass != null) {
-				if (this.repositoryCredentials == null
+				if (this.repositoryCredentials == null || this.gitCredentialsService == null
```

Comment:
This whole condition will be successful if `gitCredentialsService` is null.
In that case, the call to `showPopUpAskingToStoreCredentials` will happen and will lead to null pointer exception.